soroban-env-host/src/budget.rs

```diff
@@ -250,15 +242,35 @@ impl BudgetDimension {
     /// input, assuming all batched units have the same input size. If input
     /// is `None`, the input is ignored and the model is treated as a constant
     /// model, and amount charged is iterations * const_term.
-    pub fn charge(
+    fn charge(
         &mut self,
         ty: ContractCostType,
         iterations: u64,
         input: Option<u64>,
+        _is_cpu: bool,
     ) -> Result<(), HostError> {
         let cm = self.get_cost_model(ty);
-        let amount = cm.evaluate(input)?.saturating_mul(iterations);
-        self.counts[ty as usize] = self.counts[ty as usize].saturating_add(amount);
+        let mut amount = cm.evaluate(input)?.saturating_mul(iterations);
+
+        #[cfg(all(not(target_family = "wasm"), feature = "tracy"))]
+        if _is_cpu {
+            let _span = tracy_span!("charge");
+            _span.emit_text(ty.name());
+            _span.emit_value(amount);
+        }
+
+        if _is_cpu {
+            // Budget charging _itself_ has a nonzero overhead, on the
+            // order of 100ns or maybe 250 instructions.
+            const BUDGET_CHARGE_CPU_OVERHEAD: u64 = 250;
+            amount = amount.saturating_add(BUDGET_CHARGE_CPU_OVERHEAD);
```

Comment:
I thought of this, but then I also thought that part of why we got rid of it is that it's a bit .. recursive? like if you were to start counting the couple-hundred instructions spent, probably a lot of them are in the steps _before_ we get to this line: updating the tracker, fetching the CPU cost model and evaluating it (and then doing that again for memory), and so if we did all that a _second time_ just to "charge the cost of mesuring costs" we'd be incurring those costs again, and would need to "charge the cost of measuring costs" again, so we'd have to actually set that cost to _twice_ the measurement cost. Since this cost is not entirely negligible, I figured it might be better to just inline it inside here, where it's going to be much smaller, probably only 2-5 instructions (a branch, a saturating add, a store) and only apply to the CPU path.